### ISSUE TITLE
Featured collection links missing role

### ISSUE THREAD

**Issue by kmeleta** [Shopify/dawn#864]
**Describe the current behavior**
Featured collection empty links are just <a>

**Describe the expected behavior**
Should have aria-disabled="true" and role="link"

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 1.0.0
- Chrome Version 91.0.4472.114
- macOS Version 11.3.1

**Possible solution**

**Additional context/screenshots**
Other sections received this update in #810

**Comment by andrewetchen** [comment]
I thought I'd share some of my findings regarding this issue re: `href`'s and `href="#"` attributes for unassigned products and collections on some homepage sections.

### Context

When any of the following sections are added to the homepage, and when no product or collection is assigned, the placeholder/onboarding links are used. The HTML markup below is what gets rendered on the storefront for these links:

- **Collage**
  - Product block: `<a href="#" class="full-unstyled-link">`
  - Collection block: `<a href="" class="full-unstyled-link">`
- **Collection list**
  - Collection block: `<a href="" class="full-unstyled-link">`
- **Featured collection**
  - Onboarding/placeholder products with unassigned collection: `<a href="#" class="full-unstyled-link">`
- **Featured product**
  - Unassigned product: View full details --> `<a href="" class="link product__view-details animate-arrow">`

### Store demo links

Feel free to look at the editor link as I specifically added these sections:

- [Store](https://wegrzvgtdpb43dog-45840990230.shopifypreview.com)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127533940758/editor)

### Behaviour

When adding these sections to the homepage, placeholder products or collections are rendered. These placeholders aim to provide merchants with a _rough_ idea of how a section will look in terms of layout, formatting, etc., and potentially what parts of a section have the ability to contain links - _is it safe to assume the last point?_ 🤔 

- Could there be a better approach to imply section links?
- Perhaps we can use the [Multicolumn approach](https://screenshot.click/01-18-t4fr7-5p0lgd-.png) for these links and apply a similar style?

#### Accessibility

In terms of accessibility, @svinkle provided a great summary in #387 on how "Disabled" links shouldn't be focusable or actionable via mouse/keyboard:

>When a link is created without a destination, it's still actionable via mouse and keyboard. Activating the link refreshes the page. This may cause confusion for multiple user groups who may think activation will load a new resource.

> In order to create a disabled link which is not actionable yet retains its semantic meaning:
> 
> 1. Remove the `href` attribute. This will cause the link to not receive focus or be actionable via mouse click/Enter key.
> 2. Add the `role` attribute, setting its value to `link`. This will provide semantic meaning for assistive technology.

#### Related theme files

All of the above sections render either of the following snippets:

- [card-collection.liquid](https://github.com/Shopify/dawn/blob/main/snippets/card-collection.liquid)
- [card-product.liquid](https://github.com/Shopify/dawn/blob/main/snippets/card-product.liquid)

### Summary

Currently, "disabled" links render a, `<a href="#" class="full-unstyled-link">` or `<a href="" class="full-unstyled-link">` on the homepage.

Whichever approach we go with, I think there's an opportunity to improve the consistency of "disabled" links and aside from the homepage, perhaps there are other areas of the theme where this could be applied.

### Thoughts ?

Please feel free to add your thoughts, thanks.

**Comment by kmeleta** [comment]
Thanks Andrew. Didn't realize how many of these were cards, but as far as I'm aware we should align them all to be `<a role="link" aria-disabled="true" class="<link | full-unstyled-link>">` with the `not-allowed` cursor as Multicolumn currently is. The same a11y principle _should_ apply to all these and our motivation to inform the merchant there are empty links would be the same I'd think.

We can do a more thorough check in a PR review, but I also noticed featured product has a stray empty link as well https://screenshot.click/03-56-jzm0u-voqv1.png

**Comment by svinkle** [comment]
Agreed with Ken. As we've seen recently in live examples, placeholder cards may be published by a merchant. I'd suggest the same treatment for all "disabled" links.

**Comment by melissaperreault** [comment]
Aligned with following the same experience across 👍
